__tests__/application.spec.ts

```diff
@@ -139,4 +139,85 @@ describe('[application.ts]', () => {
       emulateOptions: application.configuration.emulateOptions
     })
   });
+
+  describe('should connect and init plugins',  () => {
+    it('should connect and init plugins with on start ', async () => {
+      // Arrange
+      const plugin = {
+        onBeforeStart: sandbox.stub().resolves()
+      };
+      const configuration: ApplicationConfig = {
+        origin: faker.internet.url(),
+        pages: [{
+          configuration: {
+            matcher: faker.random.word(),
+            name: faker.random.word(),
+          },
+          handle: {}
+        } as any],
+        emulateOptions: {
+          userAgent: faker.random.word()
+        },
+        plugins: [
+          plugin
+        ]
+      };
+
+      const routerMock = {
+        get: sandbox.stub(),
```

Comment:
I think it should be real sandbox.mock.
Because you can't prevent unexpected calls.